kubernetes-itests/src/test/java/io/fabric8/kubernetes/DeploymentIT.java

```diff
@@ -113,21 +114,41 @@ public void update() {
   @Test
   public void delete() throws InterruptedException {
     // Usually creation, deletion of things like Deployments take some time. So let's wait for a while:
-    Thread.sleep(6000);
+    await().atMost(15, TimeUnit.MINUTES).until(deploymentReady());
     assertTrue(client.extensions().deployments().inNamespace(currentNamespace).delete(deployment1));
   }
 
   @Test
   public void waitTest() throws InterruptedException {
-    Deployment deploymentOne = client.resource(deployment1).waitUntilReady(10, TimeUnit.MINUTES);
+    await().atMost(15, TimeUnit.MINUTES).until(deploymentReady());
+    Deployment deploymentOne = client.extensions().deployments()
+      .inNamespace(currentNamespace).withName("deployment1").get();
     assertTrue(Readiness.isDeploymentReady(deploymentOne));
   }
 
   @After
   public void cleanup() throws InterruptedException {
-    Thread.sleep(6000);
+    await().atMost(15, TimeUnit.MINUTES).until(deploymentReady());
```

Comment:
15 minutes would be long, isn't it? what was the previous time?

Reply:
Previously it was 10 Minutes, but it was failing, that's why increased it to 15 minutes but that is the max limit. If the condition met earlier it will proceed.